dashboard/src/components/AppView/AppView.tsx

```diff
@@ -204,20 +204,22 @@ export default function AppView() {
       cluster,
       app.installedPackageRef?.context?.namespace || "",
     );
-    setResourceRefs(parsedRefs);
+    setAppViewResourceRefs(parsedRefs);
     return () => {};
-  }, [app, cluster, kinds, resourceRefs]);
+  }, [app, cluster, kinds, appViewResourceRefs]);
 
   useEffect(() => {
-    Object.values(resourceRefs).forEach((refs: ResourceRef[]) => {
-      refs.forEach(ref => dispatch(actions.kube.getAndWatchResource(ref)));
-    });
+    if (!app || !app.installedPackageRef || !app.apiResourceRefs) {
+      return () => {};
+    }
+    // TODO(minelson): Update to watch only those resources that
+    // we're interested in watching change (deployments, statefulsets etc.)
+    // and get the rest.
+    dispatch(actions.kube.getResources(app!.installedPackageRef!, app!.apiResourceRefs, true));
```

Comment:
I guess typescript transpiler is not smart enough to know `app` is != undefined at this point

Reply:
Not sure of the sequence of events, but it appears to be happy without them now.